pandas/core/series.py

```diff
@@ -2116,7 +2122,10 @@ def map_f(values, f):
             return self._constructor(new_values,
                                      index=self.index).__finalize__(self)
         else:
-            mapped = map_f(values, arg)
+            if not_ndarray:
+                mapped = values.map(arg)
```

Comment:
I would just define ``map_f = lambda values, arg: values.map(arg)``, for ``not_ndarray``